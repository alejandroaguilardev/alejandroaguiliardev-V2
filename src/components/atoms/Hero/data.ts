export const titleData = [
	{
		title: "Hello World",
		variant: "h6",
		color: "primary.main",
		fontSize: { xs: "1rem", md: "1.5rem" },
		fontWeight: "bold",
        marginBottom:"10px"
	},
	{
		title: " Hello, I am Alejandro Aguilar",
		variant: "h1",
		color: "h1",
		fontSize: { xs: "1.5rem", md: "3rem" },
		fontWeight: "bold",
        marginBottom:"0"
	},
	{
		title: " Engineer System - Software Developer",
		variant: "h6",
		color: "text.secondary",
		fontSize: { xs: "1rem", md: "1.5rem" },
		fontWeight: "bold",
        marginBottom:"0"
	},
];
