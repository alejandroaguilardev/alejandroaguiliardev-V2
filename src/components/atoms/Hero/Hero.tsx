import Image from "next/image";
import { Grid, Typography, Box } from "@mui/material"
import { titleData } from "./data";

export const Hero = () => {
  return (
    <Grid
      container
      direction='column'
      sx={{
        background: 'url(/img/pattern.svg) no-repeat',
        justifyContent: "center",
        height: 'calc(100vh - 80px)'
      }}
    >
      <Box>
        {
          titleData.map((data) => (
            <Typography
              key={data.title}
              align='center'
              variant={data.variant === "h1" ? "h1" : "h6"}
              sx={{
                display: 'block',
                color: data.color,
                fontSize: data.fontSize,
                fontWeight: data.fontWeight,
                marginBottom:data.marginBottom
              }}>
              {data.title}
            </Typography>
          ))
        }
      </Box>
      <Image
        src='/img/alejandro-aguilar-recurse.svg'
        height={350}
        width={350}
        priority
      />
    </Grid>
  );
}
