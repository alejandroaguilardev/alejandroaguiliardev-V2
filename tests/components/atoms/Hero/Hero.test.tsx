import { Hero } from "../../../../src/components/atoms/Hero/Hero";
import { render } from "@testing-library/react";

jest.mock("lottie-react",()=>({
    useLottie: (options={})=>({View:<h1>Lottie</h1>})
}))

describe("Hero Test", () => {
	test("Hero snapshot", () => {
		const { container } = render(<Hero />);
		expect(container).toMatchSnapshot();
	});
});
