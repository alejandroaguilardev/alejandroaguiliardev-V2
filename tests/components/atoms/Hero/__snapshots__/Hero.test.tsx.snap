// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Test Hero snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-cu00av-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1jxxhvi-MuiTypography-root"
      >
        Hello World
      </h6>
      <h1
        class="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter css-1d2at2s-MuiTypography-root"
      >
         Hello, I am Alejandro Aguilar
      </h1>
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1dhpaeb-MuiTypography-root"
      >
         Engineer System - Software Developer
      </h6>
    </div>
    <h1>
      Lottie
    </h1>
  </div>
</div>
`;
